mpas_analysis/shared/io/namelist_streams_interface.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+"""
+    Module of classes / routines to manipulate fortran namelist and streams
+    files.
+
+    Phillip Wolfram
+    LANL
+    04/24/2014
+"""
+
+from subprocess import check_output, call
+from lxml import etree
+
+class NameList:
+    """
+    Class for fortran manipulation of namelist files, provides
+    read and write functionality
+    """
+
+    # constructor
+    def __init__(self, fname, readonly=True):
+        # set readonly bit
+        self.readonly = readonly
+        # input file name
+        self.fname = fname
+
+    def read(self, name):
+        # shell return value
+        return_val = check_output(['awk', '/'+name+'/{printf $3}', self.fname])
```

Comment:
As you mentioned in the PR description, it would be preferable to have pure python.  I would think it would make more sense to read in the full file on init and create a dictionary from the names and values.  Then, the various get functions (get, getint, getfloat, getbool) that I suggest we use instead of read would  would just return the  dictionary value, possibly with the appropriate casting.